packages/server/src/course/heatmap.service.ts

```diff
@@ -44,19 +52,48 @@ export class HeatmapService {
       return false;
     }
 
-    const officeHours = await OfficeHourModel.find({
-      where: { startTime: MoreThan(recent), courseId },
+    //const officeHours = await OfficeHourModel.find({
+    //  where: { startTime: MoreThan(recent), courseId },
+    //});
+    const taEvents = await EventModel.find({
+      where: { time: MoreThan(recent), courseId },
     });
 
-    if (officeHours.length === 0) {
+    if (taEvents.length === 0) {
       return false;
     }
 
     const tz = (await CourseModel.findOne({ id: courseId })).timezone;
+
+    function extractTimestamps(taEvents: EventModel[]) {
+      const hours = [];
+      taEvents.sort((a, b) => {
+        return a.time.getTime() - b.time.getTime();
+      });
+
+      let iday = taEvents[0].time.getDate();
```

Comment:
what does this do exactly? i literally do not remember what this code dose at all